src/core/control/tools/ImageHandler.cpp

```diff
@@ -85,6 +83,12 @@ auto ImageHandler::insertImage(GFile* file, double x, double y) -> bool {
     img->setWidth(width * zoom);
     img->setHeight(height * zoom);
 
+    return std::make_tuple(true, img, width * zoom, height * zoom);
+}
+
+auto ImageHandler::addImageToDocument(Image* img) -> bool {
```

Comment:
Add a boolean parameter `addInsertUndoAction` here, since that may be unwanted (say in the Lua API one may want to use only one such action for a whole bunch of images).